test/rhsmlib/services/test_register.py

```diff
@@ -249,6 +250,102 @@ def test_register_normally(self, mock_persist_consumer, mock_write_cache):
         ]
         self.assertEqual(expected_plugin_calls, self.mock_pm.run.call_args_list)
 
+    @mock.patch("rhsmlib.services.register.syspurposelib.write_syspurpose_cache", return_value=True)
+    @mock.patch("rhsmlib.services.register.managerlib.persist_consumer_cert")
+    def test_register_multiple_environment_ids(self, mock_persist_consumer, mock_write_cache):
+        self.mock_identity.is_valid.return_value = False
+        self.mock_installed_products.format_for_server.return_value = []
+        self.mock_installed_products.tags = []
+        expected_consumer = json.loads(CONSUMER_CONTENT_JSON)
+        self.mock_cp.registerConsumer.return_value = expected_consumer
+
+        register_service = register.RegisterService(self.mock_cp)
+        register_service.register("org", name="name", environments=["env-id-1", "env-id-2"])
+
+        self.mock_cp.registerConsumer.assert_called_once_with(
+            name="name",
+            facts={},
+            owner="org",
+            environments=["env-id-1", "env-id-2"],
+            environment_names=None,
+            keys=None,
+            installed_products=[],
+            jwt_token=None,
+            content_tags=[],
+            consumer_type="system",
+            role="",
+            addons=[],
+            service_level="",
+            usage="",
+        )
+        self.mock_installed_products.write_cache.assert_called()
+
+        mock_persist_consumer.assert_called_once_with(expected_consumer)
+        mock_write_cache.assert_called_once()
+        expected_plugin_calls = [
+            mock.call("pre_register_consumer", name="name", facts={}),
+            mock.call("post_register_consumer", consumer=expected_consumer, facts={}),
+        ]
+        self.assertEqual(expected_plugin_calls, self.mock_pm.run.call_args_list)
+
+    @mock.patch("rhsmlib.services.register.syspurposelib.write_syspurpose_cache", return_value=True)
+    @mock.patch("rhsmlib.services.register.managerlib.persist_consumer_cert")
+    def test_register_multiple_environment_names(self, mock_persist_consumer, mock_write_cache):
```

Comment:
If not too difficult perhaps you could consider adding a `test_register_multiple_invalid_environment_names()`

Reply:
Good point, I can add there another JSON with consumer object and use it. It should be simple.